File: Baekjoon/Others/11404.py
# import sys; input = sys.stdin.readline

def main():
    # init
    INF = 987654321
    n, m = int(input()), int(input())
    cost = [[INF for _ in range(n)] for _ in range(n)]

    for _ in range(m):
        a, b, c = map(int, input().split())
        a -= 1
        b -= 1
        cost[a][b] = min(cost[a][b], c)

    for i in range(n):
        cost[i][i] = 0

    # floyd warshall
    for k in range(n):
        for i in range(n):
            for j in range(n):
                cost[i][j] = min(cost[i][j], cost[i][k] + cost[k][j])

    # print
    answer = []
    for i in range(n):
        for j in range(n):
            if cost[i][j] == INF:
                cost[i][j] = 0
        answer.append(' '.join(map(str, cost[i])))
    print('\n'.join(answer))

if __name__ == '__main__':
    main()

File: Baekjoon/Others/11404.cpp
#include<iostream>
#include<algorithm>
#define INF 987654321
using namespace std;

void init();
void floyd();
void output();

int n, m;
int cost[100][100] = { 0, };

int main() {
	ios::sync_with_stdio(false);
	cin.tie(NULL); cout.tie(NULL);

	init();
	floyd();
	output();

	return 0;
}
void init() {
	int i, j, a, b, c;
	cin >> n >> m;
	// initialize array
	for (i = 0; i < n; i++)
	{
		for (j = 0; j < n; j++)
		{
			if (i != j)
				cost[i][j] = INF;
		}
	}

	// input cost
	for (i = 0; i < m; i++)
	{
		cin >> a >> b >> c;
		if (cost[a - 1][b - 1])
			cost[a - 1][b - 1] = min(cost[a - 1][b - 1], c);
		else
			cost[a - 1][b - 1] = c;
	}
}
void floyd() {
	int i, j, k, tmp;
	for (k = 0; k < n; k++)
	{
		for (i = 0; i < n; i++)
		{
			for (j = 0; j < n; j++)
			{
				cost[i][j] = min(cost[i][j], cost[i][k] + cost[k][j]);
			}
		}
	}

}
void output() {
	int i, j;
	// post processing and print
	for (i = 0; i < n; i++)
	{
		for (j = 0; j < n; j++)
		{
			if (cost[i][j] == INF)
				cost[i][j] = 0;
			cout << cost[i][j] << ' ';
		}
		cout << '\n';
	}
}